nsqd/nsqd.go

```diff
@@ -206,6 +212,30 @@ func (n *NSQD) GetStartTime() time.Time {
 	return n.startTime
 }
 
+// AddClient adds a client to the producers map
+func (n *NSQD) AddClient(clientID int64, client Producer) {
+	n.Lock()
+	defer n.Unlock()
```

Comment:
... but if there was another client property, from which it could be determined if the client connection had already published anything, `AddClient()` could usually be skipped